x-pack/platform/plugins/shared/sample_data_ingest/server/services/artifact_manager/utils/fetch_artifact_versions.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { DocumentationProduct, parseArtifactName, type ProductName } from '@kbn/product-doc-common';
+import * as fs from 'fs';
+import fetch from 'node-fetch';
+import * as Path from 'path';
+import { URL } from 'url';
+import { parseString } from 'xml2js';
+import { resolveLocalPath } from './resolve_local_path';
+
+type ArtifactAvailableVersions = Record<ProductName, string[]>;
+
+export const fetchArtifactVersions = async ({
+  artifactRepositoryUrl,
+}: {
+  artifactRepositoryUrl: string;
+}): Promise<ArtifactAvailableVersions> => {
+  const parsedUrl = new URL(artifactRepositoryUrl);
+
+  let xml: string;
+  if (parsedUrl.protocol === 'file:') {
```

Comment:
Question: Similar to above in `download`: consider adding validations here as well.

Reply:
Removed that code block because it looks more like a premature optimization

Reply:
Question: And just to double check, can your new validations be added here?